internal/controller/management/projectconfigs/project_configs.go

```diff
@@ -0,0 +1,311 @@
+	if err := kubeclient.PatchStatus(ctx, r.client, projectConfig, func(status *kargoapi.ProjectConfigStatus) {
+		*status = newStatus
+	}); err != nil {
+		// Prioritize the reconcile error if it exists.
+		if reconcileErr != nil {
+			logger.Error(err, "failed to update ProjectConfig status after reconciliation error")
+			return ctrl.Result{}, reconcileErr
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to update ProjectConfig status: %w", err)
+	}
+
+	// Return the reconcile error if it exists.
+	if reconcileErr != nil {
+		return ctrl.Result{}, reconcileErr
+	}
+	// Immediate requeue if needed.
+	if needsRequeue {
+		return ctrl.Result{Requeue: true}, nil
+	}
+	// Otherwise, requeue after a delay.
+	// TODO: Make the requeue delay configurable.
+	return ctrl.Result{RequeueAfter: 5 * time.Minute}, nil
+}
+
+func (r *reconciler) syncProjectConfig(
+	ctx context.Context,
+	pc *kargoapi.ProjectConfig,
+) (
+	kargoapi.ProjectConfigStatus,
+	bool,
+	error,
+) {
+	logger := logging.LoggerFromContext(ctx)
+	status := pc.Status.DeepCopy()
+
+	conditions.Set(status, &metav1.Condition{
+		Type:               kargoapi.ConditionTypeReconciling,
+		Status:             metav1.ConditionTrue,
+		Reason:             "SyncingWebhooks",
+		ObservedGeneration: pc.GetGeneration(),
+	})
+	conditions.Set(status, &metav1.Condition{
+		Type:               kargoapi.ConditionTypeReady,
+		Status:             metav1.ConditionFalse,
+		Reason:             "SyncingWebhooks",
+		ObservedGeneration: pc.GetGeneration(),
+	})
+
+	whReceivers, err := r.syncWebhookReceiversFn(ctx, pc)
+	status.WebhookReceivers = whReceivers
+	if err != nil {
+		logger.Error(err, "error syncing webhook receivers",
+			"project-config", pc.Name,
+		)
+		conditions.Set(status, &metav1.Condition{
+			Type:               kargoapi.ConditionTypeReady,
+			Status:             metav1.ConditionFalse,
+			Reason:             "SecretMisconfiguration",
+			Message:            "Failed to sync webhook receivers: " + err.Error(),
+			ObservedGeneration: pc.GetGeneration(),
+		})
+		return *status, true, err
+	}
+
+	conditions.Set(status, &metav1.Condition{
+		Type:               kargoapi.ConditionTypeReady,
+		Status:             metav1.ConditionTrue,
+		Reason:             "Synced",
+		Message:            "ProjectConfig is synced and ready for use",
+		ObservedGeneration: pc.GetGeneration(),
+	})
+	conditions.Delete(status, kargoapi.ConditionTypeReconciling)
+	return *status, false, nil
+}
+
+func (r *reconciler) syncWebhookReceivers(
+	ctx context.Context,
+	pc *kargoapi.ProjectConfig,
+) ([]kargoapi.WebhookReceiver, error) {
+	logger := logging.LoggerFromContext(ctx)
+	if pc.Spec.WebhookReceivers == nil {
+		logger.Debug("ProjectConfig does not have any receiver configurations")
+		return nil, nil
+	}
+
+	logger.Debug("syncing webhook receivers",
+		"webhook-receiver-configs", len(pc.Spec.WebhookReceivers),
+	)
+
+	var errs []error
+	var webhookReceivers []kargoapi.WebhookReceiver
```

Comment:
Because we're allowing for the possibility of moving on to setting up other receivers once an error is encountered, we don't know the exact size of the collection we're building up front, but we do know what its size would be under nominal conditions, so we can avoid a lot of unnecessary resizing by setting the initial size to zero, but the capacity to equal the length of `pc.Spec.WebhookReceivers`.

```suggestion
	webhookReceivers := make([]kargoapi.WebhookReceiver, 0, len(pc.Spec.WebhookReceivers))
```